### PR TITLE
Add snackbar and refactored code

### PR DESCRIPTION
**Pull request by Maianki**
### Files to be reviewed
- [README.md](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
- [src/App.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
- [src/components/Card/CartProductCard/CartProductCard.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-98a6c5e0ba46b35ebcadc09ba5f57174feea3608e3f7fdf680f4ab180ead3f27)
- [src/components/Card/CategoryCard/CategoryCard.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-3be2019371dc8b5afd2dd1c91807801482702ebb5efb539fb992108accce5db2)
- [src/components/Navbar/Navbar.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-547f8328c15a9a29000b7ef4690ba4f1039d148d5c208e5a48d37a2acd21982e)
- [src/components/Snackbar/Snackbar.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-45a3fb90f4dcc96613f969ef5bee5251ecf9a30ccf898186feefa1693860f510)
- [src/context/auth-context.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-50d25f7adf348217c540867a8937453cd2a126a2abdf2681a9e4a838a51f9714)
- [src/context/cart-and-wishlist-context.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-4261ae8a9a14038355bde3cee41d82c7f2c0964d77a0f2e02c798016e48710f8)
- [src/context/products-context.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-99adb1a18be68347324ac941627c9c63fbd2a51eff91da6d74bd65ca0207109e)
- [src/context/snackbar-context.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-1cc28c7dce7e0594f5a919d3ebb71d2d55f36179e27346dfe9550f51727caf7d)
- [src/pages/Profile/Profile.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-93f068f0ecc02ca613ac870d7f640244f3a4621e12401d31d34d6a28eefa608c)
- [src/pages/SignupPage/SignUpPage.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-49d14eb844b4e3736154279f406e9762af43c9ae40dd9311725a77753b47d53a)
- [src/reducers/authReducer.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-978cb6ff885969efafb050f3963f42db8388343b0f752803094897fa970c7023)
- [src/reducers/productsReducer.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-1e72764740711c06e4f3cf410674a7d5a3bb67e5236ee138c5019d857f828e3a)
- [src/reducers/snackbarReducer.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-snackbar-and-refactored-code?expand=1#diff-99b635af334a37bf68442ac5efd5353fcbdab6da5f2fec408604808bc35b599f)
### Live Preview Link
[click here for live preview ](https://shuttle-shopy-react-git-add-snackbar-and-refacto-772744-maianki.vercel.app/)

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -19,38 +22,69 @@ const AuthProvider = ({ children }) => {
...
+          type: "LOGGED_IN",
+          payload: encodedToken,
+        });
+
+        addSnackbar("You are logged in", "snackbar-primary");
+      }
     } catch (error) {
-      console.log(error);
+      const { status } = error.response;
+      if (status === 404) {
```
Comment (sainath7878) [inline src/context/auth-context.js:78]: Appreciation: Good that you're handling all status codes!

File: src/context/snackbar-context.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { createContext, useContext, useReducer } from "react";
+import { snackbarInitialState, snackbarReducer } from "../reducers";
+import { v4 as uuid } from "uuid";
+import { Snackbar } from "../components";
+
+const SnackbarContext = createContext(null);
```
Comment (sainath7878) [inline src/context/snackbar-context.js:6]: Appreciation: Good that you've made a separate context for toast!

**Review by sainath7878 — APPROVED** [review]
Everything looks good.  Code was neat and clean. Awesome work on the UI too!

**Review by vrx29 — APPROVED** [review]
Code looks good, cant find any issue. You can go ahead and merge it.

**Pull request merged**